docs/metrics.md

```diff
@@ -0,0 +1,466 @@
+| `eventstore_incoming_grpc_calls{kind="failed"}`            | [Counter](#common-types) | Total gRPC requests failed                           |
+| `eventstore_incoming_grpc_calls{kind="unimplemented"}`     | [Counter](#common-types) | Total gRPC requests made to unimplemented methods    |
+| `eventstore_incoming_grpc_calls{kind="deadline-exceeded"}` | [Counter](#common-types) | Total gRPC requests for which deadline have exceeded |
+
+Example Configuration:
+```
+"IncomingGrpcCalls": {
+	"Current": true,
+	"Total": false,
+	"Failed": true,
+	"Unimplemented": false,
+	"DeadlineExceeded": false
+}
+```
+
+Example Output:
+```
+# TYPE eventstore_current_incoming_grpc_calls gauge
+eventstore_current_incoming_grpc_calls 1 1687963622074
+
+# TYPE eventstore_incoming_grpc_calls counter
+eventstore_incoming_grpc_calls{kind="failed"} 1 1687962877623
+```
+
+#### gRPC methods
+
+In addition, EventStoreDB also records metrics for each of the gRPC methods : `StreamRead`, `StreamAppend`, `StreamBatchAppend`, `StreamDelete` and `StreamTombstone`. They are grouped together according to the mapping defined in the configuration.
+
+| Time Series | Type       | Description |
+| :---------- | :--------- | :---------- |
+| `eventstore_grpc_method_duration_seconds_bucket{activity=<LABEL>,status="successful"\|"failed",le=<DURATION>}` | [Histogram](#common-types) | Number of _LABEL_ gRPC requests that took less than or equal to _DURATION_ in seconds to process |
+
+Example Configuration:
+```
+"GrpcMethods": {
+	"StreamAppend": "append",
+	"StreamBatchAppend": "append",
+	// leaving label as blank will disable metric collection
+	"StreamRead": "",
+	"StreamDelete": "",
+	"StreamTombstone": ""
+}
+```
+
+Example Output:
+```
+# TYPE eventstore_grpc_method_duration_seconds histogram
+# UNIT eventstore_grpc_method_duration_seconds seconds
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="1E-06"} 0 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="1E-05"} 0 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="0.0001"} 129 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="0.001"} 143 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="0.01"} 168 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="0.1"} 169 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="1"} 169 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="10"} 169 1688157491029
+eventstore_grpc_method_duration_seconds_bucket{activity="append",status="successful",le="+Inf"} 169 1688157491029
+```
+
+### Kestrel
```

Comment:
maybe a link to what Kesterl is migh be good ?

Reply:
Yup. Added it!